feat(ark): implement evaluation controller with all evaluation types

skokaina opened the pull request:
## Summary

- Implement evaluation controller supporting direct, query, event, baseline, and batch modes
- Add evaluator controller for parameter management and overrides
- Core CRD definitions and webhook validation
- Fix critical annotation bug where query/event/baseline evaluations weren't receiving metadata

## Checklist

- [x] Follow the [contributor guide](./CONTRIBUTING.md)
- [x] End-to-end tests pass
- [x] Unit tests for essential parts of your code, e.g. APIs exposed via SDKs or endpoints
- [x] Update `./docs`
- [x] Recognize contributors with "@all-contributors please add <person> for <code, bug, docs, etc>"
- [x] Linked issues #eg101

skokaina committed:
feat(ark): implement evaluation controller with all evaluation types

dwmkerr commented on ark/internal/controller/evaluator_controller.go:
```diff
@@ -115,6 +144,301 @@ func (r *EvaluatorReconciler) updateStatus(ctx context.Context, evaluator arkv1a
 }
+
+// resolveEvaluatorParameters resolves evaluator parameters from various sources
+func (r *EvaluatorReconciler) resolveEvaluatorParameters(ctx context.Context, params []arkv1alpha1.Parameter, namespace string) ([]arkv1alpha1.Parameter, error) {
```
I think we might already have some common code that handles parameter resolution

skokaina replied:
The evaluator controller needs a custom implementation because it support optional parameters, however we are using the common resolver for the Adress, and custom for the Parameters due to some optional fields (keyRef..)

dwmkerr commented on ark/internal/controller/evaluator_controller.go:
```diff
@@ -115,6 +144,301 @@ func (r *EvaluatorReconciler) updateStatus(ctx context.Context, evaluator arkv1a
+// resolveParameterValue resolves a parameter value from ConfigMap sources
+func (r *EvaluatorReconciler) resolveParameterValue(ctx context.Context, valueFrom *arkv1alpha1.ValueFromSource, namespace string) (string, error) {
+	if valueFrom.ConfigMapKeyRef != nil {
+		return r.resolveConfigMapKeyRef(ctx, valueFrom.ConfigMapKeyRef, namespace)
```
That means that we shouldn't need to duplicate this

skokaina committed:
Merge remote-tracking branch 'origin' into feat/AAS-2628-eval-controller

skokaina committed:
fix: add missing genai files for evaluation support

- Add context_utils.go for evaluation context handling
- Update evaluator.go with UnifiedEvaluationRequest and CallUnifiedEvaluator
- Update agent.go, team.go, tools.go for evaluation integration
- Add operation_tracker.go updates
- Regenerate deepcopy code

skokaina committed:
fix: complete evaluation controller implementation

- Fix evaluation and evaluator webhook integration
- Register EvaluationReconciler and webhook in main.go
- Remove duplicate files and fix import issues
- Add missing controller and webhook files
- All builds now pass successfully

skokaina committed:
remove sh files

skokaina committed:
fix: resolve staticcheck linting issue with embedded ResourceValidator

- Use direct method call instead of accessing embedded field
- Fixes QF1008 staticcheck warning about removable embedded field selector

skokaina committed:
remove EVALUATION PR SUMMARY

skokaina committed:
test: skip wather test, deprecated legacy evaluation

skokaina committed:
Merge remote-tracking branch 'origin' into feat/AAS-2628-eval-controller

skokaina committed:
fix: replace FetcherExecutor with HTTPExecutor after merge

- FetcherExecutor type was renamed to HTTPExecutor in main branch
- Update GetToolType switch case to use correct type name
- Fixes build error after merging from origin/main
